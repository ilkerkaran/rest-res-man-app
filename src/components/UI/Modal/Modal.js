import React from 'react'

import './Modal.css'
import PropTypes from 'prop-types'
import Backdrop from '../Backdrop/Backdrop'

const Modal = ({ show, onClose, children }) => (
  <>
    <Backdrop
      show={show}
      onClick={onClose}
    />
    <div
      className="Modal"
      style={{
        transform: show ? 'translateY(0)' : 'translateY(-100vh)',
        opacity: show ? '1' : '0'
      }}
    >
      {children}
    </div>
  </>
)

export default Modal

Modal.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  children: PropTypes.any.isRequired,
  show: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired
}
