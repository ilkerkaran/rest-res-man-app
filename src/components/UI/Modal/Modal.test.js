/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render } from 'enzyme';
import Modal from './Modal';

describe('<Modal /> ', () => {
  const fieldProps = {
    show: true,
    onClose: jest.fn()
  };
  let wrapper;
  const Composition = () => <Modal {...fieldProps} />;

  beforeEach(() => {
    wrapper = render(<Composition />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
