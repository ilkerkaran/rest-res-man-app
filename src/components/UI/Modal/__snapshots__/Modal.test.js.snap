// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal />  should match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Backdrop",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {
        "class": "Modal",
        "style": "transform:translateY(0);opacity:1",
      },
      "children": Array [],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
        "style": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
        "style": undefined,
      },
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {
            "class": "Modal",
            "style": "transform:translateY(0);opacity:1",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "1": Object {
    "attribs": Object {
      "class": "Modal",
      "style": "transform:translateY(0);opacity:1",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "class": "Backdrop",
      },
      "children": Array [],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "class": "Backdrop",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
